### PR TITLE
Sentinelone threat file

### PR DESCRIPTION
**Pull request by r7-mgorny**
## Proposed Changes
Sentinelone threat file

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nToken: *************16c03\nrapid7/SentinelOne:3.1.0. Step name: threats_fetch_file\n",
    "meta": {},
    "output": {
      "file": {
        "content": "W3siaW5jbHVkZWQiOnRydWUsInBhdGgiOiJcXERldmljZVxcSGFyZGRpc2tWb2x1bWUyXFxVc2Vyc1xcQWRtaW5pc3RyYXRvclxcRGVza3RvcFxcdGVzdC50eHQiLCJyZWFzb24iOiIiLCJzaGExIjoiYTIxMzhmMjFlYTk2ZDk3Y2UwMGNkNTE5NzE2OTZmNzQ2NGU1ZGI4OSIsInNoYTI1NiI6IjY4Mjk2YTM3MTg4ZWUwMzA1NjY5YTIzYWNhZGUxNDg2MDVkZjMwYjgyM2Y5NTJmZDRmMzFjMGE4NTQ1NThmOTAiLCJzaXplIjo2OX1d",
        "filename": "test.txt"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:3.1.0 --debug run < tests/threats_fetch_file.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nWrong password. Password must have more than 10 characters and cannot contain whitespace.",
    "log": "Connect: Connecting...\nToken: *************16c03\nrapid7/SentinelOne:3.1.0. Step name: threats_fetch_file\nAn error occurred during plugin execution!\n\nWrong password. Password must have more than 10 characters and cannot contain whitespace.\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 311, in handle_step\n    output = self.start_step(input_message['body'], 'action', logger, log_stream, is_test, is_debug)\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 419, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.7/site-packages/sentinelone_rapid7_plugin-3.1.0-py3.7.egg/komand_sentinelone/actions/threats_fetch_file/action.py\", line 23, in run\n    assistance=\"Password must have more than 10 characters and cannot contain whitespace.\"\nkomand.exceptions.PluginException: An error occurred during plugin execution!\n\nWrong password. Password must have more than 10 characters and cannot contain whitespace.\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:3.1.0 --debug run < tests/threats_fetch_file_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nWrong password. Password must have more than 10 characters and cannot contain whitespace.",
    "log": "Connect: Connecting...\nToken: *************16c03\nrapid7/SentinelOne:3.1.0. Step name: threats_fetch_file\nAn error occurred during plugin execution!\n\nWrong password. Password must have more than 10 characters and cannot contain whitespace.\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 311, in handle_step\n    output = self.start_step(input_message['body'], 'action', logger, log_stream, is_test, is_debug)\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 419, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.7/site-packages/sentinelone_rapid7_plugin-3.1.0-py3.7.egg/komand_sentinelone/actions/threats_fetch_file/action.py\", line 23, in run\n    assistance=\"Password must have more than 10 characters and cannot contain whitespace.\"\nkomand.exceptions.PluginException: An error occurred during plugin execution!\n\nWrong password. Password must have more than 10 characters and cannot contain whitespace.\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:3.1.0 --debug run < tests/threats_fetch_file2_bad.json
</summary>
</details>

### Validate

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 2450.39ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
./komand_sentinelone/actions/activities_types/action.py:2:1: F401 '.schema.Input' imported but unused
./komand_sentinelone/actions/get_agent_details/action.py:22:29: W291 trailing whitespace
./komand_sentinelone/actions/name_available/action.py:2:1: F401 '.schema.Input' imported but unused
./komand_sentinelone/connection/connection.py:69:12: F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)
./komand_sentinelone/connection/connection.py:154:12: F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)
./komand_sentinelone/connection/connection.py:178:12: F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)
./komand_sentinelone/connection/connection.py:200:12: F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)
./komand_sentinelone/connection/tempCodeRunnerFile.py:10:12: W292 no newline at end of file
./komand_sentinelone/util/api.py:19:20: F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
98 [0.. 50.. ]
Run started:2020-09-18 20:10:01.619147

Test results:
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'password'
   Severity: Low   Confidence: Medium
   Location: ./komand_sentinelone/actions/threats_fetch_file/schema.py:12
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b105_hardcoded_password_string.html
11          ID = "id"
12          PASSWORD = "password"
13

--------------------------------------------------

Code scanned:
        Total lines of code: 4744
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 1.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 1.0
                High: 0.0
Files skipped (0):
[FAIL] Fails bandit security checks

[*] Checking for typos with Misspell...
[FAIL] Fails Misspell check
komand_sentinelone/connection/connection.py:157:28: "Repsonse" is a misspelling of **"Response"**

```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator ExceptionValidator
WARNING: Use of 'PluginException' or 'ConnectionTestException' is recommended when raising an exception.
violation: komand_sentinelone/connection/connection.py: line,  179
violation: komand_sentinelone/connection/connection.py: line,  201
violation: komand_sentinelone/connection/connection.py: line,  331
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[850]: https://example.sentinelone.com
[*] Plugin successfully validated!

----
[*] Total time elapsed: 23968.414ms

```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

File: sentinelone/plugin.spec.yaml [hunk 1]
```diff
@@ -1530,6 +1530,27 @@ actions:
...
+    input:
+      id:
+        title: ID
+        description: Threat ID
+        type: integer
+        required: true
+      password:
+        title: Password
+        description: File encryption password, min. length 10 characters and cannot contain whitespace
+        type: string
```
Comment (jschipp-r7) [inline sentinelone/plugin.spec.yaml:1545]: This needs to be `password` type to mask it in the product

File: sentinelone/plugin.spec.yaml [hunk 2]
```diff
@@ -1530,6 +1530,27 @@ actions:
...
+        required: true
+        example: "Rapid7 Insightconnect"
+    output:
+      file:
+        title: File
+        description: File of data to be imported into Anomali ThreatStream
```
Comment (jschipp-r7) [inline sentinelone/plugin.spec.yaml:1551]:
```suggestion
        description: Base64 encoded threat file
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
Comments in review and need to pass `mdl` linting

```
[*] Validating markdown...
help.md:40: MD001 Header levels should only increment by one level at a time
help.md:569: MD012 Multiple consecutive blank lines
help.md:20: MD022 Headers should be surrounded by blank lines
help.md:37: MD031 Fenced code blocks should be surrounded by blank lines
```

File: sentinelone/help.md [hunk 3]
```diff
@@ -1470,6 +1507,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 3.1.0 - Add new action Threats Fetch File
```
Comment (mrinehart-r7) [inline sentinelone/help.md:1510]:
```suggestion
* 3.1.0 - Add new action Fetch Threats File
```

File: sentinelone/plugin.spec.yaml [hunk 4]
```diff
@@ -1530,6 +1530,27 @@ actions:
         description: Return true if blacklist item was created or deleted
         type: boolean
         required: true
+  threats_fetch_file:
+    title: Threats Fetch File
```
Comment (mrinehart-r7) [inline sentinelone/plugin.spec.yaml:1534]:
```suggestion
    title: Fetch Threats File
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by jschipp-r7 — APPROVED** [review]

**Pull request merged**